<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> software | George Hall </title> <meta name="author" content="George Hall"> <meta name="description" content="My personal site! "> <meta name="keywords" content="scRNAseq, comptuer science, machine learning, artifical intelligence"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" href="/assets/css/scholar-icons.css?62b2ac103a88034e6882a5be5f3e2772"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/icons8-man-technologist-96.png?5543510b134369d982f71320fe6057cc"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://george-hall.github.io/software/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> </head> <body class="fixed-top-nav sticky-bottom-footer"> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">George</span> Hall </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item dropdown "> <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">research </a> <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown"> <a class="dropdown-item " href="/projects/postdoctoral/">postdoctoral</a> <div class="dropdown-divider"></div> <a class="dropdown-item " href="/projects/doctoral/">doctoral</a> <div class="dropdown-divider"></div> <a class="dropdown-item " href="/projects/predoctoral/">predoctoral</a> </div> </li> <li class="nav-item active"> <a class="nav-link" href="/software/">software <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">software</h1> <p class="post-description"></p> </header> <article> <p>I have a number of GitHub accounts from different times:</p> <ul> <li> <a href="https://github.com/george-hall-ucl" rel="external nofollow noopener" target="_blank">george-hall-ucl</a>: From my current, postdoctoral position. Software for the analysis of single-cell and spatial transcriptomics.</li> <li> <a href="https://github.com/george-hall-wtsi" rel="external nofollow noopener" target="_blank">george-hall-wtsi</a>: From my time at the Sanger Institute. Software to quantify complexity of large datasets.</li> <li> <a href="https://github.com/george-hall" rel="external nofollow noopener" target="_blank">george-hall</a>: Personal projects.</li> </ul> <p><br></p> <p>Some of my software, from full software packages to small scripts (each name links to the software):</p> <h4 id="dawnn"><a href="https://github.com/george-hall-ucl/dawnn" rel="external nofollow noopener" target="_blank">Dawnn</a></h4> <p><img align="right" src="../../assets/img/dawnn_logo.png" width="100"></p> <p>An R package to compute similarity between two single-cell transcriptomic samples, through a lens called <em>differential abundance</em>. Briefly, a cell population (e.g. cell type) exhibits differential abundance if it is dominated by one sample. Dawnn is described in a <a href="https://doi.org/10.1101/2023.05.05.539427" rel="external nofollow noopener" target="_blank">pre-print</a>, where we show that it is able to detect more regions of differential abundance than existing methods whilst preserving low rates of false discovery.</p> <p><br></p> <h4 id="portable-cellxgene"><a href="https://george-hall-ucl.github.io/Portable-CELLxGENE-Docs" rel="external nofollow noopener" target="_blank">Portable-CELLxGENE</a></h4> <p><img align="right" src="../../assets/img/pcxg_logo.svg" width="100"></p> <p>Software to make it easier for non-bioinformaticians to annotate their single-cell transcriptomics data using a graphical interface. Portable-CELLxGENE is a stand-alone executable version of the <a href="https://github.com/chanzuckerberg/cellxgene" rel="external nofollow noopener" target="_blank"><code class="language-plaintext highlighter-rouge">cellxgene</code></a> and <a href="https://github.com/Novartis/cellxgene-gateway" rel="external nofollow noopener" target="_blank"><code class="language-plaintext highlighter-rouge">cellxgene-gateway</code></a> Python packages, bundled together with a conda environment containing all the software that they need to run. By packaging together all required software and allowing it to be installed in a standard, graphical way, Portable-CELLxGENE allows the user to focus on analysing their data rather than solving computational problems. Described in a <a href="https://doi.org/10.48550/arXiv.2408.11844" rel="external nofollow noopener" target="_blank">pre-print</a>.</p> <p><br></p> <h4 id="spatialfeatureplotblend"><a href="https://github.com/george-hall-ucl/SpatialFeaturePlotBlend" rel="external nofollow noopener" target="_blank">SpatialFeaturePlotBlend</a></h4> <center><img src="../../assets/img/SpatialFeaturePlotBlend_example.png" height="200"></center> <p>An R function to allow the expression of two genes to be plotted in relation to one another using a two-dimensional colour scale in Seurat. This is possible with non-spatial data (using <a href="https://satijalab.org/seurat/reference/featureplot" rel="external nofollow noopener" target="_blank"><code class="language-plaintext highlighter-rouge">FeaturePlot</code></a>‚Äôs <code class="language-plaintext highlighter-rouge">blend = TRUE</code> option), but there is no such option for spatial data. This function fills that gap, and allows for more customisation of colour scales than is possible in the standard <code class="language-plaintext highlighter-rouge">FeaturePlot</code> function.</p> <p><br></p> <h4 id="rotatefeatureplot"><a href="https://gist.github.com/george-hall-ucl/c7fc1d402968fcd53bdca84678f4bb5c" rel="external nofollow noopener" target="_blank">RotateFeaturePlot</a></h4> <center><img src="../../assets/img/rotate_seurat_st_example.png" height="150"></center> <p>An R function to allow the image associated with a spatial transcriptomic sample to be rotated in a Seurat plot.</p> <p><br></p> <hr> <center><h2>Older</h2></center> <h4 id="grin"><a href="https://github.com/george-hall-wtsi/GRIn" rel="external nofollow noopener" target="_blank">GRIn</a></h4> <p>From my time at the Sanger Institute, GRIn is software to estimate characteristics of a genome from only unassembled reads. Its main application is to estimate the amount of repetitive content of a genome before its assembly, as a proxy for the difficulty its assembly will pose. It does so using only output of a k-mer counter (such as <a href="https://github.com/gmarcais/Jellyfish" rel="external nofollow noopener" target="_blank">Jellyfish</a>) from unassembled reads.</p> <p><br></p> <h4 id="silly-personal-projects">Silly personal projects</h4> <p>When putting this page together, I‚Äôve discovered a lot of funny small programming projects from about ten years ago in my personal github repository. They all suddenly ground to a halt once I began programming for a living üôÉ. I‚Äôve included them here because I‚Äôve enjoyed looking back through them, and I hope they might provide a basis for people who want to practice their programming skills.</p> <p><br></p> <h5 id="conways-game-of-life"><a href="https://github.com/george-hall/GameOfLife" rel="external nofollow noopener" target="_blank">Conway‚Äôs Game of Life</a></h5> <center><img src="../../assets/img/game_of_life_demo.gif" width="150"></center> <p>Mathematician John Conway‚Äôs <a href="https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life" rel="external nofollow noopener" target="_blank">Game of Life</a> is an example of set of rules designed to mimic life (e.g. that living things need food, space, and to reproduce). They are interesting as they often lead to emerging structure (see example above, following random initialisation). This was a fun little project!</p> <p><br></p> <h5 id="langtons-ant"><a href="https://github.com/george-hall/LangtonsAnt" rel="external nofollow noopener" target="_blank">Langton‚Äôs Ant</a></h5> <center><img src="../../assets/img/langtons_demo.gif" width="200"></center> <p>Another example of order emerging from chaos. <a href="https://en.wikipedia.org/wiki/Langton%27s_ant" rel="external nofollow noopener" target="_blank">Langton‚Äôs ‚Äúant‚Äù</a> follows a simple set of rules, such as turning left when it hits a blue tile, and right when it hits any other colour.</p> <p><br></p> <h5 id="handelplot"><a href="https://github.com/george-hall/HandelPlot" rel="external nofollow noopener" target="_blank">HandelPlot</a></h5> <center><img src="../../assets/img/handel_plot_demo.gif" width="200"></center> <p>HandelPlot is software to plot the <a href="https://en.wikipedia.org/wiki/Mandelbrot_set" rel="external nofollow noopener" target="_blank">Mandelbrot Set</a>. The Mandelbrot Set is an example of a fractal, a mathematical structure which shows complexity at all levels of magnification. It‚Äôs quite an interesting project to try to plot it: the maths behind it is relatively simple to state (requires A-Level ‚Äì i.e. pre-university ‚Äì mathematics), but the resulting complexity is very pretty. The above animation begins at the outermost point of magnification and then repeatedly zooms in, showing detail at each level. I can‚Äôt remember why I called it HandelPlot, I think maybe I planned to add music at some point?</p> </article> </div> </div> <footer class="sticky-bottom mt-5" role="contentinfo"> <div class="container"> ¬© Copyright 2025 George Hall. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js?a0db7e5d5c70cc3252b3138b0c91dcaf" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script src="/assets/js/mathjax-setup.js?a5bb4e6a542c546dd929b24b8b236dfd"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script defer src="/assets/js/progress-bar.js?2f30e0e6801ea8f5036fa66e1ab0a71a" type="text/javascript"></script> </body> </html>